### ISSUE TITLE
Type errors when using moduleResolution "NodeNext"

### ISSUE THREAD

**Issue by anmolitor** [kryops/rollup-plugin-gzip#22]
According to https://www.typescriptlang.org/docs/handbook/modules/reference.html#node16-nodenext

.ts/.tsx/.js/.jsx/.d.ts files are ES modules if the nearest ancestor package.json file contains "type": "module", otherwise CommonJS modules.

The published package in node_modules/rollup-plugin-gzip contains "type": "commonjs".
The issue with this is that typings and runtime do not match up.

```
import gzipPlugin from 'rollup-plugin-gzip';

gzipPlugin();
```

leads to:

```
This expression is not callable.
  Type 'typeof import("/.../node_modules/rollup-plugin-gzip/dist/index")' has no call signatures.ts(2349)
```

I did not open a PR to change the type to "module" because I am not sure if that breaks other projects.
What do you think is the correct solution?

**Comment by kryops** [comment]
Good question... changing to `"type": "module"` sounds like the right thing to do, CommonJS imports should still work via `exports.require`. Though it is probably best to try it out with different project configurations, and make this a major release just to be safe.

I hope I will have some time over the weekend to investigate this a bit more thoroughly.

**Comment by kryops** [comment]
I just published v4.0.0 with `type: "module"`. Successfully tested it with Node 10, so hopefully it wasn't a breaking change after all :sweat_smile:
